From: Mijo Safradin <mijo.safradin@googlemail.com>
Date: Mon, 25 Jan 2016 00:03:47 +0100
Git-commit: 1d52319118bf25ebdf02093ad1fc148ffcad5ddd
References: bsc#1087487
Subject: [PATCH] Fix race condition where new metrics are added while old
 metric still gets send to carbon server. The carbon log is reporting invalid
 metrics.

Acked-by: John Jolly <jjolly@suse.de>
---
 gmetad/gmetad.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gmetad/gmetad.c b/gmetad/gmetad.c
index 60ade92c..e04368fe 100644
--- a/gmetad/gmetad.c
+++ b/gmetad/gmetad.c
@@ -610,12 +610,12 @@ main ( int argc, char *argv[] )
          /* Sum the new values */
          hash_foreach(root.authority, do_root_summary, NULL );
 
-         /* summary completed */
-         pthread_mutex_unlock(root.sum_finished);
-
          /* Save them to RRD */
          hash_foreach(root.metric_summary, write_root_summary, NULL);
 
+         /* summary completed */
+         pthread_mutex_unlock(root.sum_finished);
+
          /* Remember our last run */
          now = apr_time_now();
          last_metadata = now;  //Updating global variable
-- 
2.13.6

